### PR TITLE
Ignore null template key values

### PR DESCRIPTION
**Pull request by jbarton123**
Currently for values set to `~` that are compiled together using this function, these values are still appended to the output. This ensures that for any null values passed into this function, that they are ignored.

### CODE REVIEW

File: src/_base/harness/config/functions.yml [hunk 1]
```diff
@@ -255,6 +255,9 @@ function('template_key_value', [template, key_value]): |
   #!php
   $output = [];
   foreach ($key_value as $key => $value) {
+    if (!$value) {
```
Comment (andytson-inviqa) [inline src/_base/harness/config/functions.yml]:
```suggestion
    if ($value===null) {
```

Comment (andytson-inviqa) [reply src/_base/harness/config/functions.yml]: '0' is a valid string but !$value is true for it

**Review by hgajjar — APPROVED** [review]

File: src/spryker/docker/image/nginx/root/etc/nginx/conf.d/zed.conf.template.twig [hunk 2]
```diff
@@ -85,4 +86,5 @@ server {
         fastcgi_param APPLICATION_STORE {{ store_code }};
     }
 }
+{% endif %}
```
Comment (andytson-inviqa) [inline src/spryker/docker/image/nginx/root/etc/nginx/conf.d/zed.conf.template.twig:89]: misses zed api further down, however there's another PR for that https://github.com/inviqa/harness-base-php/pull/685

**Review by andytson-inviqa — CHANGES_REQUESTED** [review]

File: src/_base/_twig/docker-compose.yml/service/nginx.yml.twig [hunk 3]
```diff
@@ -3,7 +3,7 @@
 {% set syncvolume = true %}
 {% endif %}
 {% set hostnames = [@('hostname')] %}
-{% set hostnames = hostnames|merge(@('hostname_aliases')|map(alias => "#{alias}." ~ @('domain'))) %}
+{% set hostnames = hostnames|merge(@('hostname_aliases')| filter(v => v is not empty) | map(alias => "#{alias}." ~ @('domain'))) %}
```
Comment (andytson-inviqa) [inline src/_base/_twig/docker-compose.yml/service/nginx.yml.twig:6]: Just this bit now that is what we want to keep after @Chemaclass's PRs now merged

Comment (andytson-inviqa) [reply src/_base/_twig/docker-compose.yml/service/nginx.yml.twig:6]: although should also be done for varnish. I'll create a PR

File: src/_base/harness/config/functions.yml [hunk 4]
```diff
@@ -255,6 +255,9 @@ function('template_key_value', [template, key_value]): |
   #!php
   $output = [];
   foreach ($key_value as $key => $value) {
+    if ($value === null) {
+      continue;
+    }
```
Comment (Chemaclass) [inline src/_base/harness/config/functions.yml:260]: Already done: https://github.com/inviqa/harness-base-php/pull/686

File: src/_base/_twig/docker-compose.yml/service/nginx.yml.twig (hunk 3 above)
Comment (andytson-inviqa) [reply src/_base/_twig/docker-compose.yml/service/nginx.yml.twig:6]: https://github.com/inviqa/harness-base-php/pull/689, I've instead done at the source

**Pull request closed without merging**